TradeItIosTicketSDK2/TradeItAlertManager.swift

```diff
@@ -79,11 +94,41 @@ import UIKit
                           withActionTitle actionTitle: String,
                           onAlertActionTapped: () -> Void = {},
                           onCancelActionTapped: (() -> Void)? = nil) {
-        let alertController = TradeItAlertProvider.provideAlert(alertTitle: title,
-                                                                alertMessage: message,
-                                                                alertActionTitle: actionTitle,
-                                                                onAlertActionTapped: onAlertActionTapped,
-                                                                onCanceledActionTapped: onCancelActionTapped)
-        viewController.presentViewController(alertController, animated: true, completion: nil)
+        add(alertClosure: {
+            let alert = TradeItAlertProvider.provideAlert(
+                alertTitle: title,
+                alertMessage: message,
+                alertActionTitle: actionTitle,
+                onAlertActionTapped: {
+                    onAlertActionTapped()
+                    self.activeAlertFinished()
+                },
+                onCanceledActionTapped: {
+                    onCancelActionTapped?()
+                    self.activeAlertFinished()
+                }
+            )
+            viewController.presentViewController(alert, animated: true, completion: nil)
+        })
+    }
+
+    func add(alertClosure alertClosure: () -> Void) {
+        alertClosureQueue.append(alertClosure)
+
+        if !self.alertBeingPresented {
+            self.displayNextAlert()
+        }
+    }
+
+    func displayNextAlert() {
+        if alertClosureQueue.count <= 0 { return }
+        let alertClosure = alertClosureQueue.removeFirst()
+        alertBeingPresented = true
+        alertClosure()
+    }
+
+    func activeAlertFinished() {
+        alertBeingPresented = false
+        displayNextAlert()
```

Comment:
This is starting to get messy. I want to make a new object and add some unit tests.